src/components/dashboard/Send.js

```diff
@@ -73,6 +77,35 @@ const ContinueButton = ({ screenProps, to, disabled, checkError }) => (
 const Send = props => {
   const [screenState, setScreenState] = useScreenState(props.screenProps)
   const [error, setError] = useState()
+  const [showDialogWithData] = useDialog()
+
+  useEffect(() => {
+    const { screenProps } = props
+
+    try {
```

Comment:
Since routeAndPathForCode is not awaiting try catch won't work. Is this catching `readCode`? If so maybe the tryCatch can be moved to a more specific place?

Reply:
You're completely right. `try/catch` is not mandatory in this case. It was there for when `routeAndPathFromCode` threw errors and wasn't returning a promise.

Reply:
done